src/domain/model/authentication/Role.ts

```diff
@@ -1,13 +1,45 @@
-import { v4 } from 'uuid';
+import { Uuid } from '../../Uuid';
+import { PermissionAssignmentAction } from './PermissionAssignmentAction';
+import { Permission } from './Permission';
+import { AssignmentActions } from './AssignmentActions';
+import { UserId } from '../user/UserId';
+import { DomainValidationError } from '../DomainValidationError';
+
+const ROLE_NAME_REG_EXP = /^[A-Z]+[A-Z_]*[A-Z]+$/;
 
 export class Role {
+  readonly permissionAssignments: AssignmentActions<Role, Permission>;
+
   constructor(
-    readonly id: RoleId,
     readonly name: string,
-    readonly creationTime: Date
-  ) {}
+    readonly creationTime: Date = new Date(),
+    permissionAssignments: PermissionAssignmentAction[] = []
```

Comment:
Would it make sense to rearrange these two params?  i.e. Might you have a scenario where you create this with assignments, or would that only happen when reading from the DB?

Reply:
Yep, this makes sense. I'll update

Reply:
Actually on a second thought I wouldn't change this. To assign permissions to the role you need to first create the role. So I would just restrict completely from the constructor.